packages/expo-sms/ios/ExpoSMSModule.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  ExpoSMSModule.swift
+//  ExpoSMS
+//
+//  Created by Alan Hughes on 11/11/2022.
+//
+
+import Foundation
+import ExpoModulesCore
+import WebKit
+import MessageUI
+import CoreServices
+import MobileCoreServices
+import UniformTypeIdentifiers
+
+struct ExpoSMSContext {
+    let promise: Promise
+    let smsDelgate: SMSDelegate
+}
+
+public class ExpoSMSModule: Module, SMSResultHandler {
+    private var smsContext: ExpoSMSContext?
+    private lazy var utils = appContext?.utilities
+
+    public func definition() -> ModuleDefinition {
+        Name("ExpoSMS")
+
+        AsyncFunction("isAvailableAsync") { (promise: Promise) in
+            promise.resolve(MFMessageComposeViewController.canSendText())
+        }
+
+        AsyncFunction("sendSMSAsync") {(addresses: [String], message: String, options: SmsOptions?, promise: Promise) in
+            sendSMSAsync(addresses: addresses, message: message, options: options, promise: promise)
+        }.runOnQueue(.main)
+    }
+
+    private func sendSMSAsync(addresses: [String], message: String, options: SmsOptions?, promise: Promise) {
+        if !MFMessageComposeViewController.canSendText() {
+            promise.reject("E_SMS_UNAVAILABLE", "SMS service not available")
+            return
+        }
+
+        if smsContext != nil {
+            promise.reject("E_SMS_SENDING_IN_PROGRESS",
+                           "Different SMS sending in progress. Await the old request and then try again.")
+            return
+        }
+
+        let smsDelgate = SMSDelegate(handler: self)
+        let context = ExpoSMSContext(promise: promise, smsDelgate: smsDelgate)
+
+        let messageComposeViewController = MFMessageComposeViewController()
+        messageComposeViewController.messageComposeDelegate = context.smsDelgate
+        messageComposeViewController.recipients = addresses
+        messageComposeViewController.body = message
+
+        if let options {
```

Comment:
`if let x` is a shorthand form of `if let x = x` that was introduced in Swift 5.7 (https://github.com/apple/swift-evolution/blob/main/proposals/0345-if-let-shorthand.md) so it compiles only on Xcode >14.0
I would still use the longer form so we don't break builds still running on Xcode 13.

Thing to consider: maybe the `options` could be non-optional? Looks like it's always called with an object on the JS side: https://github.com/expo/expo/blob/59f5e3893e70156a085736fe73005f1a4441427a/packages/expo-sms/src/SMS.ts#L74-L80

Reply:
Had a feeling I should do it the other way 😅

Reply:
Yeah, I don't think the options need to be optional, I was just mapping to the Typescript type but I can remove it